### ISSUE TITLE
NEM: Integrate OSNet-x0.25 for Person Re-Identification

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#3504]
### Objective

Add person re-identification to track individuals across cameras and time.

### Model Details

* **Model:** `torchreid/osnet_x0_25`
* **VRAM:** \~100MB
* **Priority:** MEDIUM
* **Output:** 512-dimensional embedding vector

### Implementation

**File:** `ai/enrichment/models/person_reid.py` (new file)

```python
import torchreid

class PersonReID:
    def __init__(self, model_path: str):
        self.model = torchreid.models.build_model(
            name='osnet_x0_25',
            num_classes=1000,  # Not used for feature extraction
            pretrained=True
        )
        self.model.eval()
        self.transform = torchreid.data.transforms.build_transforms(
            height=256, width=128
        )
    
    def extract_embedding(self, person_crop) -> np.ndarray:
        """Extract 512-dim embedding for person crop."""
        img = self.transform(person_crop)
        with torch.no_grad():
            embedding = self.model(img.unsqueeze(0))
        return embedding.cpu().numpy().flatten()
    
    def compute_similarity(self, emb1: np.ndarray, emb2: np.ndarray) -> float:
        """Compute cosine similarity between two embeddings."""
        return np.dot(emb1, emb2) / (np.linalg.norm(emb1) * np.linalg.norm(emb2))
    
    def is_same_person(self, emb1: np.ndarray, emb2: np.ndarray, 
                       threshold: float = 0.7) -> bool:
        """Determine if two embeddings are the same person."""
        return self.compute_similarity(emb1, emb2) > threshold

@dataclass
class ReIDResult:
    embedding: list[float]  # 512-dim
    embedding_hash: str     # For quick lookup
```

### Use Cases

* Track same person across multiple cameras
* Identify if person has been seen before
* Correlate detections across time windows

### Acceptance Criteria

- [ ] `PersonReID` class implemented
- [ ] 512-dim embedding extraction
- [ ] Similarity comparison function
- [ ] Registered with MEDIUM priority
- [ ] Unit tests with same/different person pairs
- [ ] Integration with `/enrich` endpoint
- [ ] Embedding storage strategy documented

### Design Reference

See `docs/plans/2026-01-19-model-zoo-prompt-improvements-design.md` Section 5.4
